generator/sbpg/targets/resources/sbp_messages_template.rs

```diff
@@ -89,6 +94,55 @@ impl super::SBPMessage for (((m.identifier|camel_case))) {
     fn set_sender_id(&mut self, new_id: u16) {
         self.sender_id = Some(new_id);
     }
+
+    fn into_frame(self) -> Vec<u8> {
```

Comment:
We could define a default implementation of `into_frame` method in the `SBPMessage` trait so we don't have to re-implement it in each message. But you're right that we'd still have to implement `SBPSerialize` for each message anyways.

Reply:
Never mind, it seems as though you can't give a default implementation of a "consuming" method of a trait, since the size of the `self` isn't known in the general case :(

Reply:
Ah ok yeah I knew there was probably as catch. After writing these initial comments I was thinking the framing could just be a regular old static method that has those trait requirements which the concrete implementation would call, but there is little benefits at that point other than code organization since ultimately those are generated sources.